### ISSUE TITLE
[Epic] Batch transactions before signing

### ISSUE THREAD

**Issue by katspaugh** [safe-global/web-core#1808]
**tl;dr:** allow the user combine multiple transactions into one Safe transaction before proposing it for others to sign.

[Epic doc:](https://www.notion.so/safe-global/Batched-transactions-43c5bab5a9d14a9d949832bac5b2fd9a) https://www.notion.so/safe-global/Batched-transactions-43c5bab5a9d14a9d949832bac5b2fd9a

**Designs:** tbd

## What problem are we solving?

In the Safe UI, it's possible to batch-execute Safe transactions, but it's not possible to **create** a multi-send Safe transaction consisting of Token Transfers, Settings changes, Safe App transactions etc. (It's only possible to batch-send NFTs atm).

Note: Transaction Builder allows creating complex multi-send transactions, but it's not part of the main UI.

## Why do we need batch transactions?
* They save time: you don't have to sign each individual transaction, instead you just sign one multi-send tx.
* They save money: no signatures in each tx means paying less gas
* They keep the queue clean: the batch serves as a personal tx basket, so that you can remove unwanted txs before submitting them to the queue
 * They showcase Safe's programmability as per the AA concept

## How will it look in the UI?
There will one and only one locally-stored batch queue. When creating any tx, the user will have the choice to either propose the tx immediately, or add it to his batch.

Designs and UI flows tbd, but as a quick draft:
<img width="449" alt="Screenshot 2023-03-27 at 13 43 42" src="https://user-images.githubusercontent.com/381895/227932124-58792f33-1a50-43b4-a539-54a5222b938f.png">

<img width="371" alt="Screenshot 2023-03-27 at 13 43 56" src="https://user-images.githubusercontent.com/381895/227932160-686686be-ebbb-45bd-adf1-6ee98a910ebd.png">

### Can I delete transactions for a batch?
Yes.

### What if I create a batch that has mutually exclusive transactions?
Yes, the whole batch will fail. See https://github.com/safe-global/web-core/issues/1130

**Comment by usame-algan** [comment]
Do we also want to add an option to download the Basket the same as the Transaction Builder app does (maybe even using the same .json format)?
